File: solutions/go/isogram/1/isogram.go
package isogram

import (
    "unicode"
)

func IsIsogram(word string) bool {
    wordSet := make(map[rune]bool)
	for _, r := range word {
        if unicode.IsDigit(r) {
            return false
        }
        if unicode.IsLetter(r) {
            lower := unicode.ToLower(r)
            if wordSet[lower] {
                return false
            }
            wordSet[lower] = true
        }
    }
    return true
}

File: solutions/csharp/isogram/1/Isogram.cs
public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        var seen = new HashSet<char>();
        string lowered = word.ToLower();

        foreach (char c in lowered)
        {
            if (!char.IsLetter(c))
                continue;

            if (seen.Contains(c))
                return false;
    
            seen.Add(c);
        }
        return true;
    }
}
